packages/node-http-handler/src/node-http2-handler.ts

```diff
@@ -22,50 +22,68 @@ export interface NodeHttp2HandlerOptions {
    * https://nodejs.org/docs/latest-v12.x/api/http2.html#http2_http2session_and_sockets
    */
   sessionTimeout?: number;
+
+  /**
+   * Disables sharing ClientHttp2Session instance between different HTTP/2 requests
+   * sent to the same URL. When set to true, it will create a new session instance for
+   * each request to a URL. **Default:** false.
+   * https://nodejs.org/api/http2.html#http2_class_clienthttp2session
+   */
+  disableSessionCache?: boolean;
 }
 
 export class NodeHttp2Handler implements HttpHandler {
   private readonly requestTimeout?: number;
   private readonly sessionTimeout?: number;
-  private readonly connectionPool: Map<string, ClientHttp2Session>;
+  private readonly disableSessionCache?: boolean;
+
   public readonly metadata = { handlerProtocol: "h2" };
+  private sessionList: ClientHttp2Session[];
```

Comment:
The `sessionList` stores all sessions, including the ones not present in `sessionCache`
When `disableSessionCache` is set to true, the sessions should not be stored in `sessionCache`.

Reply:
When `disableSessionCache: true`
* we're closing the session on successful response.
* we're destroying the session on reject.

So, `sessionList` is actually not required. I'll remove it.

Reply:
> So, `sessionList` is actually not required. I'll remove it.

The `sessionList` is required to store sessions, in case `destroy()` is called on handler while sessions are active.

Reply:
Decisions from off-GitHub discussions:
* rename config `disableSessionCache` to `disableConcurrentStreams`.
* convert sessionCache from `Map<string, ClientHttp2Session>()` to `Map<string, ClientHttp2Session[]>()`:
  * when disableConcurrentStreams: false, the array will have only one value.
  * when disableConcurrentStreams: true, the array will have new session per stream.